### PR TITLE
8324632: Update Zlib Data Compression Library to Version 1.3.1

### PR DESCRIPTION
**Pull request by fitzsim**
8315117: Update Zlib Data Compression Library to Version 1.3
8326351: Update the Zlib version in THIRD_PARTY_README to 1.3.1

Backport-of: 6aa721649d870354a79c3508343b14222e0edcfb

I am filing this pull request now mainly to register that I am working on this backport.  Some of the sanity checks have already failed on package download issues; I will try re-running them and otherwise investigating how to fix them.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8326351](https://bugs.openjdk.org/browse/JDK-8326351) needs maintainer approval
- [x] [JDK-8324632](https://bugs.openjdk.org/browse/JDK-8324632) needs maintainer approval
- [x] [JDK-8315117](https://bugs.openjdk.org/browse/JDK-8315117) needs maintainer approval

### Issues
 * [JDK-8324632](https://bugs.openjdk.org/browse/JDK-8324632): Update Zlib Data Compression Library to Version 1.3.1 (**Enhancement** - P4 - Approved)
 * [JDK-8315117](https://bugs.openjdk.org/browse/JDK-8315117): Update Zlib Data Compression Library to Version 1.3 (**Enhancement** - P4 - Approved)
 * [JDK-8326351](https://bugs.openjdk.org/browse/JDK-8326351): Update the Zlib version in open/src/java.base/share/legal/zlib.md to 1.3.1 (**Bug** - P4 - Approved)

### Reviewers
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)
 * [Andrew John Hughes](https://openjdk.org/census#andrew) (@gnu-andrew - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev.git pull/511/head:pull/511` \
`$ git checkout pull/511`

Update a local copy of the PR: \
`$ git checkout pull/511` \
`$ git pull https://git.openjdk.org/jdk8u-dev.git pull/511/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 511`

View PR using the GUI difftool: \
`$ git pr show -t 511`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/511.diff">https://git.openjdk.org/jdk8u-dev/pull/511.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk8u-dev/pull/511#issuecomment-2147960098)

### CODE REVIEW

**Comment by fitzsim** [comment]
Other than path differences and license changes in zlib.md in 11 versus THIRD_PARTY_README in 8, the changes applied cleanly.

**Comment by fitzsim** [comment]
I guess the Debian `s390x` and `ppc64le` sanity check jobs could be updated from `buster` to `bullseye`, since these URLs 404:

http://httpredir.debian.org/debian/dists/buster/main/binary-s390x/Packages
http://httpredir.debian.org/debian/dists/buster/main/binary-ppc64el/Packages

whereas these exist:

http://httpredir.debian.org/debian/dists/bullseye/main/binary-s390x/Packages.gz
http://httpredir.debian.org/debian/dists/bullseye/main/binary-ppc64el/Packages.gz

**Comment by fitzsim** [comment]
For the Windows `x86` failures, this URL 404s:

https://debian.fmi.uni-sofia.bg/~aangelov/VS2010Express1.iso

This seems like a good substitute:

https://archive.org/download/vs-2010-express-1/VS2010Express1.iso

I confirmed the SHA256 hash matches the value expected by the environment variable in the script, `a9d5dcdf55e539a06547a8ebbc63d55dc167113e09ee9e42096ab9098313039b`.

**Comment by fitzsim** [comment]
The archive.org download is slow, and 700 MiB is a lot to pull from archive.org on every job run.  So caching this file somewhere closer (network-wise) to the builder would be nice.  However, from my reading about GHA caches, they are per-branch, so the file would be redownloaded once per pull request anyway.

**Comment by fitzsim** [comment]
Darn, I forgot to change the PR title to "Backport 6aa721649d870354a79c3508343b14222e0edcfb"; I just changed it now.  I will see if the bot notices and adds the "backport" tag, otherwise I will look into the SKARA backport command, or look at labelling it manually.

**Comment by fitzsim** [comment]
/approval 8324632 request This is an almost-clean backport from 11 and passes jdk_tier1 zip tests.

**Comment by fitzsim** [comment]
/approval 8315117 request This is an almost-clean backport from 11 and passes jdk_tier1 zip tests.

**Comment by fitzsim** [comment]
/approval 8326351 request This is an almost-clean backport from 11 and passes jdk_tier1 zip tests.

File: THIRD_PARTY_README [hunk 1]
```diff
@@ -3043,14 +3043,14 @@ performance, or use of this material.
 
 -------------------------------------------------------------------------------
 
-%% This notice is provided with respect to zlib v1.2.13, which may be included 
+%% This notice is provided with respect to zlib v1.3.1, which may be included
```
Comment (jerboaa) [inline THIRD_PARTY_README:3046]: There is a trailing space after `included`. Please remove.

File: THIRD_PARTY_README [hunk 2]
```diff
@@ -3043,14 +3043,14 @@ performance, or use of this material.
...
 -------------------------------------------------------------------------------
 
-%% This notice is provided with respect to zlib v1.2.13, which may be included 
+%% This notice is provided with respect to zlib v1.3.1, which may be included 
 with JRE 8, JDK 8, and OpenJDK 8.
 
 --- begin of LICENSE ---
 
-  version 1.2.13, October 13th, 2022
+  version 1.3.1, October 13th, 2022
```
Comment (jerboaa) [inline THIRD_PARTY_README:3051]: 1.3.1 got released on January 22, 2024. We should use that date.

**Review by jerboaa — CHANGES_REQUESTED** [review]
For JDK 8 there are other `THIRD_PARTY_README` files which would need the same treatment than the top-level-one:

```
$ find -name THIRD_PARTY_README | grep -v build/
./jaxp/THIRD_PARTY_README
./corba/THIRD_PARTY_README
./nashorn/THIRD_PARTY_README
./langtools/THIRD_PARTY_README
./hotspot/THIRD_PARTY_README
./jaxws/THIRD_PARTY_README
./THIRD_PARTY_README
./jdk/THIRD_PARTY_README
```

I'd suggest to fix the top level one and then apply the same patch to others. Note that the previous update of zlib to 1.2.13 didn't include the other THIRD_PARTY_README  changes.

**Comment by fitzsim** [comment]
> Note that the previous update of zlib to 1.2.13 didn't include the other THIRD_PARTY_README changes.

Darn, that's why I missed it; I grep'd for "1.2.13", sorry.  Will fix.

File: THIRD_PARTY_README (hunk 1 above)
Comment (fitzsim) [reply THIRD_PARTY_README]: Done.  There is trailing whitespace throughout the THIRD_PARTY_README files, which I left alone, but given that I was changing this line anyway, it is a good opportunity to remove this occurrence.

File: THIRD_PARTY_README (hunk 2 above)
Comment (fitzsim) [reply THIRD_PARTY_README]: Fix pushed.

**Comment by fitzsim** [comment]
> For JDK 8 there are other `THIRD_PARTY_README` files which would need the same treatment as the top-level-one:

Fix pushed.

**Comment by fitzsim** [comment]
Thank you for reviewing.

File: jdk/THIRD_PARTY_README [hunk 3]
```diff
@@ -3043,12 +3043,12 @@ performance, or use of this material.
...
-%% This notice is provided with respect to zlib v1.2.11, which may be included 
+%% This notice is provided with respect to zlib v1.3.1, which may be included
 with JRE 8, JDK 8, and OpenJDK 8.
 
 --- begin of LICENSE ---
 
-  version 1.2.11, January 15th, 2017
+  version 1.3.1, January 22, 2024
 
   Copyright (C) 1995-2017 Jean-loup Gailly and Mark Adler
```
Comment (jerboaa) [inline jdk/THIRD_PARTY_README:3053]: This needs an update like the top-level one too: `1995-2024`

Comment (fitzsim) [reply jdk/THIRD_PARTY_README]: Oops, pushed fix.  Confirmed that all eight `THIRD_PARTY_README` files are identical now:

```
$ md5sum THIRD_PARTY_README corba/THIRD_PARTY_README hotspot/THIRD_PARTY_README jaxp/THIRD_PARTY_README jaxws/THIRD_PARTY_README jdk/THIRD_PARTY_README langtools/THIRD_PARTY_README nashorn/THIRD_PARTY_README 
05a09b12b11038ec75df8529f41df140  THIRD_PARTY_README
05a09b12b11038ec75df8529f41df140  corba/THIRD_PARTY_README
05a09b12b11038ec75df8529f41df140  hotspot/THIRD_PARTY_README
05a09b12b11038ec75df8529f41df140  jaxp/THIRD_PARTY_README
05a09b12b11038ec75df8529f41df140  jaxws/THIRD_PARTY_README
05a09b12b11038ec75df8529f41df140  jdk/THIRD_PARTY_README
05a09b12b11038ec75df8529f41df140  langtools/THIRD_PARTY_README
05a09b12b11038ec75df8529f41df140  nashorn/THIRD_PARTY_README

```

**Comment by jerboaa** [comment]
The `CAInterop.java#teliasonerarootcav1` test failure is: https://bugs.openjdk.org/browse/JDK-8333654

**Review by jerboaa — APPROVED** [review]
LGTM

**Comment by fitzsim** [comment]
> LGTM

Thank you for reviewing; sorry for the THIRD_PARTY_README tedium, I will know for next time to keep them all identical.

I already requested approval, so I will watch for that, then issue the integrate command.

**Comment by fitzsim** [comment]
Hmm, the "rfr" label is still there; I will watch to make sure the bot removes that....

**Comment by jerboaa** [comment]
> OK, thank you for filing that. Has it been seen on master, 21, etc., or just 8?

JDK head (currently 23) and probably all earlier releases.

**Comment by jerboaa** [comment]
> Hmm, the "rfr" label is still there; I will watch to make sure the bot removes that....

My understanding of the `rfr` label is that it's mirroring the GH comments to the mailing lists. That's it.

**Comment by jerboaa** [comment]
@fitzsim Please merge master. This should have some of the test failures fixed and probably will only have `sslroot*` cacerts issues left. Tracked in https://bugs.openjdk.org/browse/JDK-8333788 Thanks!

**Comment by fitzsim** [comment]
Oh, misinterpreted "merge master" as a request to integrate; will try to get the tests rerun later today.

**Comment by fitzsim** [comment]
OK, merged master, tests are running.

**Comment by fitzsim** [comment]
Looks like the bot removed the `sponsor` label post-merge; I re-issued the `integrate` command.

**Review by gnu-andrew — APPROVED** [review]
Backport looks good. Thanks for duplicating the change to all the `THIRD_PARTY_README` files. It appears only the one in jdk makes the final image, so once this change is in, I'll check that they are all actually the same and remove the duplicates to spare future changes this duplication.

In 9, they were removed by [JDK-8169925](https://bugs.openjdk.org/browse/JDK-8169925) which replaced them with the `.md` legal files. They made some sense in 8u when there were multiple Mercurial subrepositories but everything has been in one Git repository for over two years now.

**Comment by gnu-andrew** [comment]
> @fitzsim To use the `/backport` command, you need to be in the OpenJDK [census](https://openjdk.org/census) and your GitHub account needs to be linked with your OpenJDK username ([how to associate your GitHub account with your OpenJDK username](https://wiki.openjdk.org/display/skara#Skara-AssociatingyourGitHubaccountandyourOpenJDKusername)).

This command is used on commits to automatically generate a backport. You don't need it if you have already created the backport yourself. It is unlikely to work for 8u anyway, due to the different paths.

**Comment by gnu-andrew** [comment]
> /approval 8324632 request This is an almost-clean backport from 11 and passes jdk_tier1 zip tests.

You should be able to leave out the bug ID and just issue this once.

**Pull request closed without merging**